### ISSUE TITLE
Deserialize method should not return `unit`

### ISSUE THREAD

**Issue by knocte** [joemphilips/DotNetLightning#87]
The consequence of this is that the types need their members to be `mutable`, which is ugly.

**Comment by joemphilips** [comment]
Yes, I totally agree. But right now I suppose the priority to refactor it is low.

If there are any particular example that it is doing harm, I like to know about it.
